### PR TITLE
Back to Callback

### PR DESCRIPTION
**Pull request by rpiaggio**
Use `Callback` again instead of `SyncIO`, now that we have a `scalajs-react` bundle that uses `CallbackTo` for sync effects and `IO` for async.

### CODE REVIEW

File: common/src/main/scala/explore/model/AppContext.scala [hunk 1]
```diff
@@ -106,38 +107,38 @@ case class AppContext[F[_]](
...
-        f(logger.warn(message))
-      def info(message: => String): SyncIO[Unit]                =
-        f(logger.info(message))
-      def debug(message: => String): SyncIO[Unit]               =
-        f(logger.debug(message))
-      def trace(message: => String): SyncIO[Unit]               =
-        f(logger.trace(message))
-    }
-  }
+  // val syncLogger: Logger[DefaultS] = {
```
Comment (toddburnside) [inline common/src/main/scala/explore/model/AppContext.scala:117]: Commented code?

File: explore/src/main/scala/explore/HelpBody.scala [hunk 2]
```diff
@@ -136,11 +136,11 @@ object HelpBody {
         }
       }
       .componentDidMount { $ =>
-        implicit val ctx = $.props.ctx
+        // implicit val ctx = $.props.ctx
```
Comment (toddburnside) [inline explore/src/main/scala/explore/HelpBody.scala:139]: More commented code.

File: common/src/main/scala/explore/model/AppContext.scala (hunk 1 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/model/AppContext.scala:117]: Thx for catching these :+1:

**Review by toddburnside — APPROVED** [review]
LGTM - but I don't think tree.txt is needed.

**Comment by rpiaggio** [comment]
> LGTM - but I don't think tree.txt is needed.

That one slipped in :D Removed in latest commit.

**Pull request merged**